internal/txapp/txapp.go

```diff
@@ -384,6 +429,26 @@ func (r *TxApp) Finalize(ctx context.Context, blockHeight int64) (appHash []byte
 		return nil, nil, err
 	}
 
+	// Execute state modifications for the hard forks that activate at this
+	// height. These changes are associated with other consensus logic or
+	// parameters changes, otherwise a resolution might be more sensible.
+	for _, fork := range r.Activations(blockHeight) {
+		if fork.StateMod == nil {
+			continue
+		}
+		r.log.Infof("running StateMod for %q", fork.Name)
+		if err := fork.StateMod(ctx, &common.App{
```

Comment:
Sort've unfortunate we have to do this twice (once for consensus params, once for everything else), but seems probably unavoidable.